aidial_adapter_vertexai/chat/gemini/adapter.py

```diff
@@ -242,6 +245,28 @@ async def set_finish_reason(
         await consumer.set_finish_reason(openai_reason)
 
 
+async def create_attachments_from_citations(
+    response: GenerationResponse,
+    consumer: Consumer,
+    citations_limit: int = DEFAULT_CITATIONS_LIMIT,
+) -> None:
+    if response.candidates is None or not len(response.candidates):
+        return None
+    citation_metadata = response.candidates[0].citation_metadata
+
+    if (
+        citation_metadata is None
+        or citation_metadata.citations is None
+        or not len(citation_metadata.citations)
+    ):
+        return None
+
+    for citation in citation_metadata.citations[:citations_limit]:
+        await consumer.add_attachment(
+            Attachment(url=citation.uri, title=citation.title)
```

Comment:
When I tested, this `citation.title` field was always empty

Maybe we should add some custom title like `Citation source`?

Reply:
If it title is empty, let's use the url itself as a title.
See how it's done in the playground: https://github.com/epam/ai-dial-adapter-vertexai/issues/85
We don't need to be smarter than them.